### ISSUE TITLE
Segmentation fault when asking a new question while the model is generating previous answer

### ISSUE THREAD

**Issue by gallegi** [nrl-ai/llama-assistant#11]
It seems like there is a problem with thread management.
While the model have not finishing generating answer to the previous question, I add a new prompt. Then it might freeze or crash with segmentation fault error

**Comment by vietanhdev** [comment]
@gallegi  Does it happen anymore?

**Comment by gallegi** [comment]
Yeah it is still happening to me. You haven't got this issue, have you ?
